### PR TITLE
Update dependences and build

### PR DESCRIPTION
**Pull request by anselmbradford**
## Changes

- Updates build gulp tasks to ES6 syntax.
- Replaces uglifier with `uglifyjs-webpack-plugin` to match cfgov-refresh configuration.
- Updates `babel-preset-env`, `deepmerge`, `finalhandler`, `run-sequence`, `serve-static`, `sinon`, `wcag`, `webpack`.

## Testing

- `gulp build && gulp test` should pass.

### CODE REVIEW

File: scripts/gulp/scripts.js [hunk 1]
```diff
@@ -7,7 +7,24 @@ const gulpIgnore = require( 'gulp-ignore' );
 const gulpRename = require( 'gulp-rename' );
 const webpack = require( 'webpack' );
 const webpackStream = require( 'webpack-stream' );
-const vinylNamed = require('vinyl-named');
+const UglifyWebpackPlugin = require( 'uglifyjs-webpack-plugin' );
+const vinylNamed = require( 'vinyl-named' );
```
Comment (jimmynotjim) [inline scripts/gulp/scripts.js]: Do you need this plugin?

Comment (anselmbradford) [reply scripts/gulp/scripts.js]: It's used https://github.com/cfpb/capital-framework/commit/2681c26429f5b1b76175192c636e6b29193c75dc#diff-66f9adfefaa465f6da5be9394caa2259R34 @contolini ?

Comment (contolini) [reply scripts/gulp/scripts.js]: Yes, I believe it's used to name the files `cf-whatever.js`. See https://github.com/shama/vinyl-named#example

Comment (jimmynotjim) [reply scripts/gulp/scripts.js]: Ahhh, right. Carry on.

**Pull request merged**